### PR TITLE
feat: Alert support id and ref

### LINKED ISSUE #48334: Alert cannot support id and ref
```jsx
<Alert ref={ref} id="alert_id" />
```

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- close #48334

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | feat: Alert support id and ref |
| 🇨🇳 Chinese | feat: Alert 组件支持 id 和 ref |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/alert/Alert.tsx [hunk 1]
```diff
@@ -130,7 +133,6 @@ const Alert: React.FC<AlertProps> = (props) => {
     warning.deprecated(!closeText, 'closeText', 'closable.closeIcon');
   }
 
-  const ref = React.useRef<HTMLDivElement>(null);
```
Comment (li-jia-nan) [inline components/alert/Alert.tsx:133]: 这个 ref 发现之前没有用到，直接删了

File: components/alert/Alert.tsx [hunk 2]
```diff
@@ -102,7 +104,7 @@ const CloseIconNode: React.FC<CloseIconProps> = (props) => {
   ) : null;
 };
 
-const Alert: React.FC<AlertProps> = (props) => {
+const Alert = React.forwardRef<HTMLDivElement, AlertProps>((props, ref) => {
```
Comment (zombieJ) [inline components/alert/Alert.tsx:107]: `ref` 不是 HTMLDivElement，应该是 ref.nativeElement

Comment (zombieJ) [reply components/alert/Alert.tsx:107]: 另外，没有测试用例

File: components/alert/Alert.tsx (hunk 1 above)
Comment (li-jia-nan) [reply components/alert/Alert.tsx:133]: done

File: components/alert/Alert.tsx (hunk 2 above)
Comment (li-jia-nan) [reply components/alert/Alert.tsx]: done~

**Review by afc163 — DISMISSED** [review]

File: components/alert/index.ts [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
 
-import type { AlertProps } from './Alert';
+import type { AlertProps, AlertRef } from './Alert';
 import InternalAlert from './Alert';
 import ErrorBoundary from './ErrorBoundary';
 
 export type { AlertProps } from './Alert';
 
-type CompoundedComponent = React.FC<AlertProps> & {
+type CompoundedComponent = React.ForwardRefExoticComponent<
```
Comment (zombieJ) [inline components/alert/index.ts:9]: 原本的写的也不太好，这个 typeof 直接获取就够了，不用重新实现：

```tsx
type CompoundedComponent = typeof InternalAlert & { ... }
```

Comment (li-jia-nan) [reply components/alert/index.ts]: 我看其它组件里面也有类似的写法，感觉可以都改成 `typeof xxxx & { ... }` 这样

Comment (li-jia-nan) [reply components/alert/index.ts]: - #48346

**Review by zombieJ — APPROVED** [review]

**Pull request merged**